tools/chapel-py/chplcheck/rules.py

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright 2020-2023 Hewlett Packard Enterprise Development LP
+# Copyright 2004-2019 Cray Inc.
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import chapel
+from chapel.core import *
+import re
+
+def name_for_linting(node):
+    name = node.name()
+    if name.startswith("chpl_"):
+        name = name.removeprefix("chpl_")
+    return name
+
+def check_camel_case(node):
+    return re.fullmatch(r'_?([a-z]+([A-Z][a-z]+|\d+)*|[A-Z]+)\$?', name_for_linting(node))
+
+def check_pascal_case(node):
+    return re.fullmatch(r'_?(([A-Z][a-z]+|\d+)+|[A-Z]+)\$?', name_for_linting(node))
```

Comment:
nit, I believe dollar signs can appear anywhere in an ident (although they are deprecated, so not that important)

Reply:
I'll just replace dollar signs with lowercase letters, which seems to have the desired behavior in terms of camelCase and PascalCase checking.